access-token-management/src/AccessTokenManagement/AccessTokenHandler.cs

```diff
@@ -70,11 +70,14 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             {
                 logger.AccessTokenHandlerAuthenticationFailed();
                 metrics.AccessTokenAuthenticationFailed(token.ClientId, TokenRequestType);
+
+                return response;
             }
 
-            return response;
+            dPoPNonce = response.GetDPoPNonce();
```

Comment:
I think there was a bug in the original AccessTokenHandler

https://github.com/DuendeSoftware/foss/blob/0add3ccdf05458c7655be8812d2dc4f9121e53f3/access-token-management/src/AccessTokenManagement/AccessTokenHandler.cs

If there was a dpop nonce error, it would retry with the dpop nonce. But then it wouldn't store the nonce, it would immediately return the response. This is now fixed.

Reply:
sure, but hopefully this method will be phased out in a future version in favor of the new extensibility model.